File: Framework Demo/objc/AugmentifySDK Demo/SDKLogs.m
//
//  SDKLogs.m
//  AugmentifySDK Demo
//
//  Created by Adrian König Mintellity on 18.07.19.
//  Copyright © 2019 Mintellity. All rights reserved.
//

#import "SDKLogs.h"

@implementation SDKLogs


@end

#ifdef AugmentifyImport
@implementation SDKLogs(AugmentifyExtension)

- (void)didReceiveError:(NSError * _Nonnull)error {
    NSLog(@"%@",@"-----ErrorLog-----");
    NSLog(@"%@",error);
    NSLog(@"%@",@"-----ErrorLog-----");
}

- (void)syncChangedFromStateWithOldState:(enum AugmentifySyncState)oldState newState:(enum AugmentifySyncState)newState {
    
//    NSLog(@"%@",@"-----SyncState-----");
    switch (newState){
            
        case AugmentifySyncStateNotStarted:
            NSLog(@"AugmentifySyncStateNotStarted");
            break;
        case AugmentifySyncStateError:
            NSLog(@"AugmentifySyncStateError");
            break;
        case AugmentifySyncStateStarting:
            NSLog(@"AugmentifySyncStateStarting");
            break;
        case AugmentifySyncStateAllLoaded:
            NSLog(@"AugmentifySyncStateAllLoaded");
            break;
    }
//    NSLog(@"%@",@"-----SyncState-----");
}

@end
#endif

File: Cocoapods Demo/swift/AugmentifySDK Swift Demo/SDKLogs.swift
//
//  SDKLogs.swift
//  AugmentifySDK Swift Demo
//
//  Created by Adrian König Mintellity on 18.07.19.
//  Copyright © 2019 Mintellity. All rights reserved.
//

import Foundation
import AugmentifySDK

class DemoSDKErrorLog : AugmentifyErrorDelegate {
    func didReceiveError(_ error: Error) {
        print("-----ErrorLog-----")
        print(error)
        print("-----ErrorLog-----")
    }
}

class DemoSDKLog : AugmentifyStatusDelegate{
    func syncChangedFromState(oldState: AugmentifySyncState, newState: AugmentifySyncState) {
        print("-----SyncState-----")
        print("\(oldState.toString) -> \(newState.toString)")
        print("-----SyncState-----")
    }
}
